# HG changeset patch
# User Sean Farley <sean@mcs.anl.gov>
# Date 1332262784 18000
#      Tue Mar 20 11:59:44 2012 -0500
# Node ID 18830790deb4dd8857f93427ba7794f81709665d
# Parent  8220d6713ea21d8a119911ceed844e89f9fbcca4
parmetis: fix bug reported by jfettig; '<' to '<=' in xyzpart

diff -r 8220d6713ea2 -r 18830790deb4 libparmetis/xyzpart.c
--- a/libparmetis/xyzpart.c	Wed Mar 21 11:00:11 2012 -0500
+++ b/libparmetis/xyzpart.c	Tue Mar 20 11:59:44 2012 -0500
@@ -109,11 +109,11 @@ void IRBinCoordinates(ctrl_t *ctrl, grap
     /* get into a iterative backet boundary refinement */
     for (l=0; l<5; l++) {
       /* determine bucket counts */
       iset(nbins, 0, lcounts);
       for (j=0, i=0; i<nvtxs;) {
-        if (cand[i].key < emarkers[j+1]) {
+        if (cand[i].key <= emarkers[j+1]) {
           lcounts[j]++;
           i++;
         }
         else {
           j++;
@@ -156,11 +156,11 @@ void IRBinCoordinates(ctrl_t *ctrl, grap
       rcopy(nbins+1, nemarkers, emarkers);
     }
 
     /* assign the coordinate to the appropriate bin */
     for (j=0, i=0; i<nvtxs;) {
-      if (cand[i].key < emarkers[j+1]) {
+      if (cand[i].key <= emarkers[j+1]) {
         bxyz[cand[i].val*ndims+k] = j;
         i++;
       }
       else {
         j++;
